File: src/client/java/set/starlev/starredheltix/sound/ModSounds.java
package set.starlev.starredheltix.sound;

import net.minecraft.registry.Registries;
import net.minecraft.registry.Registry;
import net.minecraft.sound.SoundEvent;
import net.minecraft.util.Identifier;

public class ModSounds {
    public static final SoundEvent VOTING_REMINDER = registerSoundEvent("voting_reminder");
    public static final SoundEvent UPDATE_AVAILABLE = registerSoundEvent("update_available");
    
    private static SoundEvent registerSoundEvent(String name) {
        Identifier id = Identifier.of("starredheltix", name);
        return Registry.register(Registries.SOUND_EVENT, id, SoundEvent.of(id));
    }
    
    public static void registerSounds() {
        // Sounds are registered when the class is loaded
    }
}

File: src/main/kotlin/set/starlev/ModSounds.kt
package set.starlev

import net.minecraft.core.Registry
import net.minecraft.core.registries.BuiltInRegistries
import net.minecraft.resources.ResourceLocation
import net.minecraft.sounds.SoundEvent

object ModSounds {
    lateinit var VOTING_REMINDER: SoundEvent
    lateinit var UPDATE_AVAILABLE: SoundEvent

    fun register() {
        VOTING_REMINDER = registerSoundEvent("voting_reminder")
        UPDATE_AVAILABLE = registerSoundEvent("update_available")
    }

    private fun registerSoundEvent(name: String): SoundEvent {
        val id = ResourceLocation.fromNamespaceAndPath("starredheltix", name)
        val soundEvent = SoundEvent.createVariableRangeEvent(id)
        Registry.register(BuiltInRegistries.SOUND_EVENT, id, soundEvent)
        return soundEvent
    }
}